services/brig/src/Brig/API/User.hs

```diff
@@ -1166,43 +1209,74 @@ lookupProfile self other =
 -- Otherwise only the 'PublicProfile' is accessible for user 'self'.
 -- If 'self' is an unknown 'UserId', return '[]'.
 lookupProfiles ::
+  ( MonadUnliftIO m,
+    MonadClient m,
+    MonadReader Env m,
+    MonadLogger m,
+    MonadMask m,
+    MonadHttp m,
+    HasRequestId m
+  ) =>
   -- | User 'self' on whose behalf the profiles are requested.
   Local UserId ->
   -- | The users ('others') for which to obtain the profiles.
   [Qualified UserId] ->
-  ExceptT FederationError (AppIO r) [UserProfile]
+  ExceptT FederationError m [UserProfile]
 lookupProfiles self others =
   concat
     <$> traverseConcurrentlyWithErrors
       (lookupProfilesFromDomain self)
       (bucketQualified others)
 
 lookupProfilesFromDomain ::
-  Local UserId -> Qualified [UserId] -> ExceptT FederationError (AppIO r) [UserProfile]
+  ( MonadClient m,
+    MonadReader Env m,
+    MonadLogger m,
+    MonadMask m,
+    MonadHttp m,
+    HasRequestId m
+  ) =>
+  Local UserId ->
+  Qualified [UserId] ->
+  ExceptT FederationError m [UserProfile]
 lookupProfilesFromDomain self =
   foldQualified
     self
     (lift . lookupLocalProfiles (Just (tUnqualified self)) . tUnqualified)
     lookupRemoteProfiles
 
-lookupRemoteProfiles :: Remote [UserId] -> ExceptT FederationError (AppIO r) [UserProfile]
+lookupRemoteProfiles ::
+  ( MonadIO m,
+    MonadReader Env m,
+    MonadLogger m
+  ) =>
+  Remote [UserId] ->
+  ExceptT FederationError m [UserProfile]
 lookupRemoteProfiles (qUntagged -> Qualified uids domain) =
   Federation.getUsersByIds domain uids
 
 -- FUTUREWORK: This function encodes a few business rules about exposing email
 -- ids, but it is also very complex. Maybe this can be made easy by extracting a
 -- pure function and writing tests for that.
 lookupLocalProfiles ::
+  forall m.
+  ( MonadClient m,
+    MonadReader Env m,
+    MonadLogger m,
+    MonadMask m,
+    MonadHttp m,
+    HasRequestId m
+  ) =>
   -- | This is present only when an authenticated user is requesting access.
   Maybe UserId ->
   -- | The users ('others') for which to obtain the profiles.
   [UserId] ->
-  (AppIO r) [UserProfile]
+  m [UserProfile]
 lookupLocalProfiles requestingUser others = do
-  users <- wrapClient (Data.lookupUsers NoPendingInvitations others) >>= mapM userGC
+  users <- (Data.lookupUsers NoPendingInvitations others) >>= mapM userGC
   css <- case requestingUser of
-    Just localReqUser -> toMap <$> wrapClient (Data.lookupConnectionStatus (map userId users) [localReqUser])
-    Nothing -> mempty
+    Just localReqUser -> toMap <$> Data.lookupConnectionStatus (map userId users) [localReqUser]
+    Nothing -> pure $ toMap [] -- mempty
```

Comment:
I had to change this in order to get the type to derive properly, but we should change it back.